# Chapter 1 - The Nature of Econometrics and Economic Data

**Introductory Econometrics** by Jeffrey M. Wooldridge (5th Edition)

---

This notebook contains solutions to computer exercises from Chapter 1. These exercises focus on exploring and understanding economic data using descriptive statistics.

## Setup

Import necessary libraries and load utility functions.

In [1]:
# Standard imports
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Custom utilities
import sys
sys.path.append('..')
from utils import (
    load_wooldridge_data,
    summary_statistics,
    plot_correlation_matrix
)

# Display settings
pd.set_option('display.max_columns', None)
pd.set_option('display.precision', 4)
%matplotlib inline

print("Setup complete!")

Setup complete!


---

## Computer Exercise C1

**Problem Statement:**

Use the data in WAGE1.RAW for this exercise.

(i) Find the average education level in the sample. What are the lowest and highest years of education?

(ii) Find the average hourly wage in the sample. Does it seem high or low?

(iii) The wage data are reported in 1976 dollars. Using the Economic Report of the President (2011 or later), obtain and report the Consumer Price Index (CPI) for the years 1976 and 2010.

(iv) Use the CPI values from part (iii) to find the average hourly wage in 2010 dollars. Now does the average hourly wage seem reasonable?

(v) How many women are in the sample? How many men?

**Dataset:** wage1

### Data Loading and Initial Exploration

In [2]:
# Load the WAGE1 dataset (will auto-download if not found)
wage1 = load_wooldridge_data('wage1')

# Display first few rows
print("First 5 rows of WAGE1 dataset:")
print(wage1.head())

print(f"\nDataset shape: {wage1.shape}")
print(f"Number of observations: {len(wage1)}")

# Get column information
print(f"\nColumns in dataset: {list(wage1.columns)}")

CSV file not found for wage1. Downloading...
Successfully downloaded and converted to: d:\Users\achyu\git\learnStats\Econometrics\Practice\notebooks\..\data\wage1.csv
Loaded wage1 from d:\Users\achyu\git\learnStats\Econometrics\Practice\notebooks\..\data\wage1.csv
Shape: (526, 24)
First 5 rows of WAGE1 dataset:
   wage  educ  exper  tenure  nonwhite  female  married  numdep   smsa  \
0  3.10    11      2       0     False    True    False       2   True   
1  3.24    12     22       2     False    True     True       3   True   
2  3.00    11      2       0     False   False    False       2  False   
3  6.00     8     44      28     False   False     True       0   True   
4  5.30    12      7       2     False   False     True       1  False   

   northcen  south  west  construc  ndurman  trcommpu  trade  services  \
0     False  False  True     False    False     False  False     False   
1     False  False  True     False    False     False  False      True   
2     False  False  

### Part (i): Education Statistics

In [3]:
# Part (i): Find average, minimum, and maximum education levels
avg_education = wage1['educ'].mean()
min_education = wage1['educ'].min()
max_education = wage1['educ'].max()

print("Education Statistics:")
print(f"Average education level: {avg_education:.2f} years")
print(f"Minimum education level: {min_education} years")
print(f"Maximum education level: {max_education} years")

# Additional statistics for context
print(f"\nAdditional education statistics:")
print(f"Standard deviation: {wage1['educ'].std():.2f} years")
print(f"Median education: {wage1['educ'].median():.1f} years")

Education Statistics:
Average education level: 12.56 years
Minimum education level: 0 years
Maximum education level: 18 years

Additional education statistics:
Standard deviation: 2.77 years
Median education: 12.0 years


### Part (ii): Average Hourly Wage

In [4]:
# Part (ii): Find average hourly wage (in 1976 dollars)
avg_wage_1976 = wage1['wage'].mean()

print("Wage Statistics (1976 dollars):")
print(f"Average hourly wage: ${avg_wage_1976:.2f}")
print(f"Minimum wage: ${wage1['wage'].min():.2f}")
print(f"Maximum wage: ${wage1['wage'].max():.2f}")
print(f"Standard deviation: ${wage1['wage'].std():.2f}")

# Assessment: Does it seem high or low?
print(f"\nAssessment:")
print(f"The average wage of ${avg_wage_1976:.2f} per hour in 1976 dollars seems relatively")
print(f"reasonable for that time period, but we need to adjust for inflation to make")
print(f"a proper assessment against today's standards.")

Wage Statistics (1976 dollars):
Average hourly wage: $5.90
Minimum wage: $0.53
Maximum wage: $24.98
Standard deviation: $3.69

Assessment:
The average wage of $5.90 per hour in 1976 dollars seems relatively
reasonable for that time period, but we need to adjust for inflation to make
a proper assessment against today's standards.


### Part (iii): Consumer Price Index Data

In [5]:
# Part (iii): Consumer Price Index values
# Source: Economic Report of the President
# Base year: 1982-84 = 100

cpi_1976 = 56.9  # CPI for 1976
cpi_2010 = 218.1  # CPI for 2010

print("Consumer Price Index (CPI) Values:")
print(f"1976 CPI: {cpi_1976}")
print(f"2010 CPI: {cpi_2010}")
print(f"Base year: 1982-84 = 100")

# Calculate inflation factor
inflation_factor = cpi_2010 / cpi_1976
print(f"\nInflation factor from 1976 to 2010: {inflation_factor:.2f}")
print(f"This means prices increased by {(inflation_factor - 1) * 100:.1f}% from 1976 to 2010")

Consumer Price Index (CPI) Values:
1976 CPI: 56.9
2010 CPI: 218.1
Base year: 1982-84 = 100

Inflation factor from 1976 to 2010: 3.83
This means prices increased by 283.3% from 1976 to 2010


### Part (iv): Inflation-Adjusted Wages

In [6]:
# Part (iv): Convert 1976 wages to 2010 dollars
avg_wage_2010 = avg_wage_1976 * inflation_factor

print("Wage Comparison:")
print(f"Average wage in 1976 dollars: ${avg_wage_1976:.2f}")
print(f"Average wage in 2010 dollars: ${avg_wage_2010:.2f}")

print(f"\nAssessment:")
print(f"The average wage of ${avg_wage_2010:.2f} per hour in 2010 dollars")
print(f"seems reasonable, as it's above the federal minimum wage of $7.25")
print(f"that was in effect in 2010. This represents purchasing power")
print(f"equivalent to what ${avg_wage_1976:.2f} could buy in 1976.")

Wage Comparison:
Average wage in 1976 dollars: $5.90
Average wage in 2010 dollars: $22.60

Assessment:
The average wage of $22.60 per hour in 2010 dollars
seems reasonable, as it's above the federal minimum wage of $7.25
that was in effect in 2010. This represents purchasing power
equivalent to what $5.90 could buy in 1976.


### Part (v): Gender Distribution

In [7]:
# Part (v): Count women and men in the sample
# The 'female' variable is an indicator (1 = female, 0 = male)
num_women = wage1['female'].sum()
num_men = len(wage1) - num_women
total_observations = len(wage1)

print("Gender Distribution:")
print(f"Number of women: {num_women}")
print(f"Number of men: {num_men}")
print(f"Total observations: {total_observations}")

# Calculate percentages
pct_women = (num_women / total_observations) * 100
pct_men = (num_men / total_observations) * 100

print(f"\nPercentages:")
print(f"Women: {pct_women:.1f}%")
print(f"Men: {pct_men:.1f}%")

Gender Distribution:
Number of women: 252
Number of men: 274
Total observations: 526

Percentages:
Women: 47.9%
Men: 52.1%


---

## Computer Exercise C2

**Problem Statement:**

Use the data in BWGHT.RAW to answer this question.

**Dataset:** bwght


In [9]:
# Test the new CSV-based data loader for BWGHT
import importlib
import sys
sys.path.append('..')
from utils import econometrics_utils
importlib.reload(econometrics_utils)
from utils import load_wooldridge_data

# Load the BWGHT dataset with new CSV approach (downloads directly as CSV)
print("Testing new CSV-based data loader for BWGHT...")
bwght = load_wooldridge_data('bwght', force_download=True)

print("BWGHT Dataset Overview:")
print(f"Shape: {bwght.shape}")
print(f"Columns: {list(bwght.columns)}")
print(f"\nData types:")
for col, dtype in bwght.dtypes.items():
    print(f"  {col}: {dtype}")

print("\nFather's education analysis:")
print(f"Total observations: {len(bwght)}")
print(f"Missing values: {bwght['fatheduc'].isna().sum()}")
print(f"Valid values: {bwght['fatheduc'].notna().sum()}")
print(f"Zero values: {(bwght['fatheduc'] == 0).sum()}")

print("\nFirst 5 rows:")
print(bwght.head())

Testing new CSV-based data loader for BWGHT...
CSV file not found for bwght. Downloading...
Successfully downloaded and converted to: d:\Users\achyu\git\learnStats\Econometrics\Practice\notebooks\..\data\bwght.csv
Loaded bwght from d:\Users\achyu\git\learnStats\Econometrics\Practice\notebooks\..\data\bwght.csv
Shape: (1388, 14)
BWGHT Dataset Overview:
Shape: (1388, 14)
Columns: ['faminc', 'cigtax', 'cigprice', 'bwght', 'fatheduc', 'motheduc', 'parity', 'male', 'white', 'cigs', 'lbwght', 'bwghtlbs', 'packs', 'lfaminc']

Data types:
  faminc: float64
  cigtax: float64
  cigprice: float64
  bwght: float64
  fatheduc: float64
  motheduc: float64
  parity: float64
  male: boolean
  white: boolean
  cigs: float64
  lbwght: float64
  bwghtlbs: float64
  packs: float64
  lfaminc: float64

Father's education analysis:
Total observations: 1388
Missing values: 196
Valid values: 1192
Zero values: 0

First 5 rows:
   faminc  cigtax  cigprice  bwght  fatheduc  motheduc  parity   male  white  \
0    

**(i) How many women are in the sample, and how many report smoking during 
pregnancy?**

In [10]:
# Part (i): Sample size and smoking
total_women = len(bwght)
smokers = bwght['cigs'].gt(0).sum()
print(f"Total women in sample: {total_women}")
print(f"Women who smoked during pregnancy: {smokers}")
print(f"Percentage who smoked: {(smokers/total_women)*100:.1f}%")

Total women in sample: 1388
Women who smoked during pregnancy: 212
Percentage who smoked: 15.3%


**(ii) What is the average number of cigarettes smoked per day? Is the average a good measure of the "typical" woman in this case? Explain.**

In [11]:
# Part (ii): Average cigarettes per day
avg_cigs_all = bwght['cigs'].mean()
print(f"Average cigarettes per day (all women): {avg_cigs_all:.2f}")

Average cigarettes per day (all women): 2.09


The average is not a good measure as the proportion of women who smoke is only 15.3%, meaning 15.3% of the population would account for nearly 100% of the total cigarettes smoked which does not represent the entire population

**(iii) Among women who smoked during pregnancy, what is the average number of cigarettes smoked per day? How does this compare with your answer from part (ii), and why?**

In [12]:
# Part (iii): Average among smokers only
smokers_only = bwght[bwght['cigs'] > 0]
avg_cigs_smokers = smokers_only['cigs'].mean()
print(f"Average cigarettes per day (smokers only): {avg_cigs_smokers:.2f}")

Average cigarettes per day (smokers only): 13.67


If one aims to note the average count of cigarettes smoked per day by smokers, then conditioning the sample space to include only smokers would provide a more useful statistic than the entire sample's as seen in (ii)

**(iv) Find the average of fatheduc in the sample. Why are only 1,192 observations used to compute this average?**

In [13]:
# Part (iv): Father's education (with clean CSV data)
total_observations = len(bwght)
missing_fatheduc = bwght['fatheduc'].isna().sum()
valid_fatheduc = bwght['fatheduc'].notna().sum()
avg_fatheduc = bwght['fatheduc'].mean()

print(f"Total observations in dataset: {total_observations}")
print(f"Missing (NaN) values for father's education: {missing_fatheduc}")
print(f"Valid observations for father's education: {valid_fatheduc}")
print(f"Average father's education: {avg_fatheduc:.2f} years")

Total observations in dataset: 1388
Missing (NaN) values for father's education: 196
Valid observations for father's education: 1192
Average father's education: 13.19 years


There are only 1192 ovservations as there exist missing values in the data set where the particular sample did not state the fatheduc value

**(v) Report the average family income and its standard deviation in dollars.**

In [14]:
# Part (v): Family income
avg_income = bwght['faminc'].mean()
std_income = bwght['faminc'].std()
print(f"Average family income: ${avg_income:.0f}")
print(f"Standard deviation: ${std_income:.0f}")

Average family income: $29
Standard deviation: $19


---

## Computer Exercise C3: Michigan School Data (MEAP01)

**Problem Statement:**

The data in MEAP01 are for the state of Michigan in the year 2001. Use these data to answer the following questions.

**Dataset:** meap01

In [15]:
# Test the new CSV-based data loader for MEAP01
import importlib
import sys
sys.path.append('..')
from utils import econometrics_utils
importlib.reload(econometrics_utils)
from utils import load_wooldridge_data

# Load the MEAP01 dataset with new CSV approach (downloads directly as CSV)
print("Testing new CSV-based data loader for MEAP01...")
meap01 = load_wooldridge_data('meap01', force_download=True)

print("MEAP01 Dataset Overview:")
print(f"Shape: {meap01.shape}")
print(f"Columns: {list(meap01.columns)}")
print(f"\nData types:")
for col, dtype in meap01.dtypes.items():
    print(f"  {col}: {dtype}")
print("\nFirst 5 rows:")
print(meap01.head())

Testing new CSV-based data loader for MEAP01...
CSV file not found for meap01. Downloading...
Successfully downloaded and converted to: d:\Users\achyu\git\learnStats\Econometrics\Practice\notebooks\..\data\meap01.csv
Loaded meap01 from d:\Users\achyu\git\learnStats\Econometrics\Practice\notebooks\..\data\meap01.csv
Shape: (1823, 11)
MEAP01 Dataset Overview:
Shape: (1823, 11)
Columns: ['dcode', 'bcode', 'math4', 'read4', 'lunch', 'enroll', 'expend', 'exppp', 'lenroll', 'lexpend', 'lexppp']

Data types:
  dcode: float64
  bcode: float64
  math4: float64
  read4: float64
  lunch: float64
  enroll: float64
  expend: float64
  exppp: float64
  lenroll: float64
  lexpend: float64
  lexppp: float64

First 5 rows:
    dcode   bcode  math4  read4  lunch  enroll     expend     exppp  lenroll  \
0  1010.0  4937.0   83.3   77.8  40.60   468.0  2747475.0  5870.673   6.1485   
1  2070.0   597.0   90.3   82.3  27.10   679.0  1505772.0  2217.632   6.5206   
2  2080.0  4860.0   61.9   71.4  41.75   400

**(i) Find the largest and smallest values of math4. Does the range make sense? Explain.**

In [16]:
# Part (i): Range of math4 scores
math4_min = meap01['math4'].min()
math4_max = meap01['math4'].max()
math4_range = math4_max - math4_min

print(f"Math4 minimum: {math4_min}")
print(f"Math4 maximum: {math4_max}")
print(f"Math4 range: {math4_range}")

Math4 minimum: 0.0
Math4 maximum: 100.0
Math4 range: 100.0


Yes, we are told that the math4 is the percentage of 4th grades students earning satisfactory scores in math


**(ii) How many schools have a perfect pass rate on the math test? What percentage is this of the total sample?**

In [17]:
# Part (ii): Perfect pass rates
total_schools = len(meap01)
perfect_math = (meap01['math4'] == 100).sum()
perfect_math_pct = (perfect_math / total_schools) * 100

print(f"Total schools: {total_schools}")
print(f"Schools with perfect math pass rate: {perfect_math}")
print(f"Percentage with perfect pass rate: {perfect_math_pct:.2f}%")

Total schools: 1823
Schools with perfect math pass rate: 38
Percentage with perfect pass rate: 2.08%


**(iii) How many schools have math pass rates of exactly 50%?**

In [18]:
# Part (iii): Schools with 50% pass rate
fifty_percent_math = (meap01['math4'] == 50).sum()

print(f"Schools with exactly 50% math pass rate: {fifty_percent_math}")

Schools with exactly 50% math pass rate: 17


**(iv) Compare the average pass rates for the math and reading scores. Which test is harder to pass?**

In [20]:
# Part (iv): Compare math and reading averages
avg_math4 = meap01['math4'].mean()
avg_read4 = meap01['read4'].mean()

print(f"Average math4 pass rate: {avg_math4:.2f}%")
print(f"Average read4 pass rate: {avg_read4:.2f}%")
print(f"Difference (math - reading): {avg_math4 - avg_read4:.2f} percentage points")

Average math4 pass rate: 71.91%
Average read4 pass rate: 60.06%
Difference (math - reading): 11.85 percentage points


We see that the reading pass rate is lower indicating that the reading test is harder

**(v) Find the correlation between math4 and read4. What do you conclude?**

In [21]:
# Part (v): Correlation between math and reading
correlation = meap01['math4'].corr(meap01['read4'])

print(f"Correlation between math4 and read4: {correlation:.4f}")

Correlation between math4 and read4: 0.8427


A correlation of 0.824 may tell us that students who excel in one tend to excel in the other

**(vi) The variable exppp is expenditure per pupil. Find the average of exppp along with its standard deviation. Would you say there is wide variation in per pupil spending?**

In [22]:
# Part (vi): Per pupil expenditure statistics
avg_exppp = meap01['exppp'].mean()
std_exppp = meap01['exppp'].std()
min_exppp = meap01['exppp'].min()
max_exppp = meap01['exppp'].max()

print(f"Average expenditure per pupil: ${avg_exppp:.2f}")
print(f"Standard deviation: ${std_exppp:.2f}")
print(f"Minimum expenditure: ${min_exppp:.2f}")
print(f"Maximum expenditure: ${max_exppp:.2f}")
print(f"Coefficient of variation: {(std_exppp/avg_exppp)*100:.1f}%")

Average expenditure per pupil: $5194.87
Standard deviation: $1091.89
Minimum expenditure: $1206.88
Maximum expenditure: $11957.64
Coefficient of variation: 21.0%


A CV of 21% indicates moderately high variation in pupil spending. 

**(vii) Suppose School A spends $6,000 per student and School B spends $5,500 per student. By what percentage does School A's spending exceed School B's? Compare this to 100 · [log(6,000) – log(5,500)], which is the approximation percentage difference based on the difference in the natural logs.**

In [23]:
# Part (vii): Percentage difference calculations
import numpy as np

school_a_spending = 6000
school_b_spending = 5500

# Exact percentage difference
exact_pct_diff = ((school_a_spending - school_b_spending) / school_b_spending) * 100

# Log approximation
log_approx = 100 * (np.log(school_a_spending) - np.log(school_b_spending))

print(f"School A spending: ${school_a_spending}")
print(f"School B spending: ${school_b_spending}")
print(f"Exact percentage difference: {exact_pct_diff:.2f}%")
print(f"Log approximation: {log_approx:.2f}%")
print(f"Difference between methods: {abs(exact_pct_diff - log_approx):.2f} percentage points")

School A spending: $6000
School B spending: $5500
Exact percentage difference: 9.09%
Log approximation: 8.70%
Difference between methods: 0.39 percentage points




---

## Computer Exercise C4: Job Training Data (JTRAIN2)

**Note:** Complete this section after downloading the JTRAIN2 dataset.


### Data Loading and Initial Exploration

In [24]:
# Load the JTRAIN2 dataset
jtrain2 = load_wooldridge_data('jtrain2')

print("JTRAIN2 Dataset Overview:")
print(f"Shape: {jtrain2.shape}")
print(f"Columns: {list(jtrain2.columns)}")
print("\nFirst 5 rows:")
print(jtrain2.head())

CSV file not found for jtrain2. Downloading...
Successfully downloaded and converted to: d:\Users\achyu\git\learnStats\Econometrics\Practice\notebooks\..\data\jtrain2.csv
Loaded jtrain2 from d:\Users\achyu\git\learnStats\Econometrics\Practice\notebooks\..\data\jtrain2.csv
Shape: (445, 19)
JTRAIN2 Dataset Overview:
Shape: (445, 19)
Columns: ['train', 'age', 'educ', 'black', 'hisp', 'married', 'nodegree', 'mosinex', 're74', 're75', 're78', 'unem74', 'unem75', 'unem78', 'lre74', 'lre75', 'lre78', 'agesq', 'mostrn']

First 5 rows:
   train   age  educ  black   hisp  married  nodegree  mosinex  re74  re75  \
0   True  37.0  11.0   True  False     True      True     13.0   0.0   0.0   
1   True  22.0   9.0  False   True    False      True     13.0   0.0   0.0   
2   True  30.0  12.0   True  False    False     False     13.0   0.0   0.0   
3   True  27.0  11.0   True  False    False      True     13.0   0.0   0.0   
4   True  33.0   8.0   True  False    False      True     13.0   0.0   0.0   

### Part (i): Training Program Participation
**Use the indicator variable train to determine the fraction of men receiving job 
training.**

In [25]:
# Part (i): Fraction of men receiving job training
fraction_trained = jtrain2['train'].mean()
print(f"Fraction of men receiving job training: {fraction_trained:.4f}")

Fraction of men receiving job training: 0.4157


### Part (ii): Earnings Comparison by Training Status
**The variable re78 is earnings from 1978, measured in thousands of 1982 dollars. 
Find the averages of re78 for the sample of men receiving job training and the 
sample not receiving job training. Is the difference economically large?**

In [ ]:
# Part (ii): Compare 1978 earnings between training groups
avg_re78_trained = jtrain2[jtrain2['train'] == 1]['re78'].mean()
avg_re78_not_trained = jtrain2[jtrain2['train'] == 0]['re78'].mean()
print(f"Average 1978 earnings (trained): ${avg_re78_trained:.2f}")
print(f"Average 1978 earnings (not trained): ${avg_re78_not_trained:.2f}")

Average 1978 earnings (trained): $6.35
Average 1978 earnings (not trained): $4.55


We see that the untrained earner makes only around 71% the earnings as one who is trained in 1978. This is an economically large difference as training is expected to increase earnigns by around 40%.

### Part (iii): Unemployment Rates by Training Status
**The variable unem78 is an indicator of whether a man is unemployed or not in 
1978. What fraction of the men who received job training are unemployed? What 
about for men who did not receive job training? Comment on the difference.**

In [29]:
# Part (iii): Compare 1978 unemployment rates between training groups
unemp_rate_trained = jtrain2[jtrain2['train'] == 1]['unem78'].mean()
unemp_rate_not_trained = jtrain2[jtrain2['train'] == 0]['unem78'].mean()
print(f"1978 unemployment rate (trained): {unemp_rate_trained:.2f}")
print(f"1978 unemployment rate (not trained): {unemp_rate_not_trained:.2f}")

1978 unemployment rate (trained): 0.24
1978 unemployment rate (not trained): 0.38


Untrained men are expected to be unemployed at a rate around 58% higher than their trained counterparts, indicating that those without job training find it more difficult to recieve employment.

### Part (iv): Program Effectiveness Assessment
**From parts (ii) and (iii), does it appear that the job training program was effective? 
What would make our conclusions more convincing?**

It seems that the job training program was effective when taking a cursory look at the numbers. A two sample t-test can determine if (ii) is statistically significant, and a two sample z-test can determine if (iii) is statistically significant.

---

## Computer Exercise C5: Fertility Data (FERTIL2)

**Problem Statement:**

The data in FERTIL2.DTA were collected on women living in the Republic of Botswana in 1988. The variable children refers to the number of living children. The variable electric is a binary indicator equal to one if the woman's home has electricity, and zero if not.

**Dataset:** fertil2

### Data Loading and Initial Exploration

In [30]:
# Load the FERTIL2 dataset
fertil2 = load_wooldridge_data('fertil2')

print("FERTIL2 Dataset Overview:")
print(f"Shape: {fertil2.shape}")
print(f"Columns: {list(fertil2.columns)}")
print("\nFirst 5 rows:")
print(fertil2.head())

CSV file not found for fertil2. Downloading...
Successfully downloaded and converted to: d:\Users\achyu\git\learnStats\Econometrics\Practice\notebooks\..\data\fertil2.csv
Loaded fertil2 from d:\Users\achyu\git\learnStats\Econometrics\Practice\notebooks\..\data\fertil2.csv
Shape: (4361, 27)
FERTIL2 Dataset Overview:
Shape: (4361, 27)
Columns: ['mnthborn', 'yearborn', 'age', 'electric', 'radio', 'tv', 'bicycle', 'educ', 'ceb', 'agefbrth', 'children', 'knowmeth', 'usemeth', 'monthfm', 'yearfm', 'agefm', 'idlnchld', 'heduc', 'agesq', 'urban', 'urbeduc', 'spirit', 'protest', 'catholic', 'frsthalf', 'educ0', 'evermarr']

First 5 rows:
   mnthborn  yearborn   age  electric  radio     tv  bicycle  educ  ceb  \
0       5.0      64.0  24.0      True   True   True     True  12.0  0.0   
1       1.0      56.0  32.0      True   True   True     True  13.0  3.0   
2       7.0      58.0  30.0      True  False  False    False   5.0  1.0   
3      11.0      45.0  42.0      True  False   True    False   

### Part (i): Children Statistics


**(i) Find the smallest and largest values of children in the sample. What is the average of children?**


In [31]:
# Part (i): Find min, max, and average number of children
min_children = fertil2['children'].min()
max_children = fertil2['children'].max()
avg_children = fertil2['children'].mean()
print(f"Minimum number of children: {min_children}")
print(f"Maximum number of children: {max_children}")
print(f"Average number of children: {avg_children:.2f}")


Minimum number of children: 0.0
Maximum number of children: 13.0
Average number of children: 2.27


### Part (ii): Electricity Access


**(ii) What percentage of women have electricity in the home?**


In [34]:
# Part (ii): Percentage of women with electricity
num_with_electricity = fertil2['electric'].sum()
total_women = len(fertil2)
pct_with_electricity = (num_with_electricity / total_women) * 100
print(f"Percentage of women with electricity: {pct_with_electricity:.2f}% per hour with electricity: {pct_with_electricity:.2f}%")

Percentage of women with electricity: 14.01% per hour with electricity: 14.01%


### Part (iii): Children by Electricity Status


**(iii) Compute the average of children for those without electricity and do the same for those with electricity. Comment on what you find.**


In [35]:
# Part (iii): Average children for women with and without electricity
avg_children_no_electricity = fertil2[fertil2['electric'] == 0]['children'].mean()
avg_children_with_electricity = fertil2[fertil2['electric'] == 1]['children'].mean()
print(f"Average number of children (no electricity): {avg_children_no_electricity:.2f}")
print(f"Average number of children (with electricity): {avg_children_with_electricity:.2f}")

Average number of children (no electricity): 2.33
Average number of children (with electricity): 1.90


The households without electricity are expected to have more children than households with electricity.

### Part (iv): Causality Assessment


**(iv) From part (iii), can you infer that having electricity "causes" women to have fewer children? Explain.**

I do not think there exists a causal effect due to confounding variables such as geography, economic status, industry of earners and others. For example a household on a rural farm may not have electricity and has more children to help on the farm.

---

## Notes and Key Takeaways

### Important Concepts from Chapter 1:

1. **Descriptive Statistics**: Mean, median, standard deviation, min/max provide insights into data distribution
2. **Inflation Adjustment**: Always adjust monetary values for inflation when comparing across time periods
3. **Missing Data**: Be aware of missing values and how they affect sample sizes in calculations
4. **Misleading Averages**: When data has many zeros, the overall average may not represent the "typical" non-zero case
5. **Data Exploration**: Always examine your data before running regressions

### Key Formulas:

- **Inflation Adjustment**: Real Value = Nominal Value × (CPI_target / CPI_original)
- **Percentage Change**: % Change = [(New - Old) / Old] × 100
- **Log Approximation**: % Change ≈ 100 × [log(New) - log(Old)]

---

## References

- Wooldridge, J. M. (2013). *Introductory Econometrics: A Modern Approach* (5th ed.). Cengage Learning.
- Dataset source: https://faculty.utrgv.edu/diego.escobari/teaching/Datasets.html
- CPI data: Economic Report of the President (various years)